README.md

```diff
@@ -1,5 +1,124 @@
 # PostgreSQL controller
 
+This is a Kubernetes controller for managing users and their access rights to a PostgreSQL database instance.
+Its purpose is to make a codified description of what users have access to what databases and for what reason along with providing an auditable log of changes.
+
+# Design
+
+The controller will handle user and database management on PostgreSQL instances with two Kubernetes [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
+
+## Databases
+
+The CRD `PostgreSQLDatabase` specified details about a database on a specific instance.
+It is scoped as cluster wide ie. has no namespace.
+
+The main purpose of this resource is to create databases on a specific host with a specific name used by application services.
+Instances of `PostgreSQLUser` can then give a specific developer access to the database by referenceing the name.
+
+```yaml
+apiVersion: lunar.bank/v1beta1
+kind: PostgreSQLDatabase
+metadata:
+  name: user
+spec:
+  name: user
+  password:
+    value: <strong-password>
+  host:
+    value: some.host.com
+```
+
+The `password` and `host` fields are modelled like Kubernetes core's [`EnvVarSource`](https://github.com/kubernetes/api/blob/665c8a257c1af277521b08dd43d5c73570405ef0/core/v1/types.go#L1847-L1862), so it can specify raw values like above or reference ConfigMaps and Secrets.
```

Comment:
Maybe we shouldn't allow raw values for password? And require a `secretKeyRef`?

Reply:
To let users specify what to use I'd like to keep it this way. In tests it is nice to be able to just run it with a raw values as well.